### DISCUSSION TITLE
Support for XR809

### DISCUSSION THREAD

**Discussion by sanchosk** [arendst/Tasmota#10992] [Q&A] [answered]
Hi, all.
By accident I've bought wifi door/window sensors running on XR809. I wanted to buy Bluetooth version and clicked on wrong link :)
Now, I have 4 wifi sensors that work only with TuYa environment - whenever they are triggered, they contact some amazon cloud server to report values.
The chip inside seems to have open SDK (https://github.com/XradioTech/XR809SDK).
Is there any interest to add Tasmota support towards such chip?
I am thinking of de-soldering it from the board and solder ESP12 on the position, but need to do wiring checks first ;)

**Comment by barbudor** [comment, marked as answer]
Brain transplant is for sure the best way and many people have done similar.
If you need help, do not hesitate to publish photos.

Porting to other MCU is a very long task and neither this one or RTL chips are in considerations for now
We start to see more ESP32 in devices and with new ESP-C3 coming, focusing on Espressif seems to be the best.

**Reply by sanchosk** [reply]
Agree. I reversed the black magick (not sure how they step up the voltage to 3.3V using one part that looks like transistor and one coil), the rest is pretty simple - voltage divider for battery voltage measurement, push button with pull-up resistor and RC filter for the reed switch... Even the pins add up to ESP12, just need to get rid of the bottom GND pin that would prevent working with flash memory. Then adjusting actions to ensure the wake up is on reset and not pin 6 and enable pin is not grounded...
Worst thing - I have exactly 0 ESP12 chips and don't really want to de-solder Wemos boards ;)
I'll grab some photos and draw schematics tomorrow.

**Comment by openshwprojects** [comment]
I have also recently stumbled upon this XR809 door sensor and after some time I managed to create my own firmware for it.

My XR809 firmware has basic compatibility with Tasmota HTTP protocol and I successfully tested it with HTTAS Home Assistant plugin. I can now use my XR809 sensor to create automations in Home assistant, opening a door can trigger lights, etc.

Here's a detailed guide on XR809 SDK setup, compilation process and short description of how I created my firmware:
https://www.elektroda.com/rtvforum/topic3806769.html#19447386

I might put up my firmware for XR809 on Github. Also, has anyone found XR809 in some other devices? So far I know also about LCD temperature sensor based on XR809. If you found any other XR devicses, let me know!

**Reply by EnDeRWiGiN** [reply]
There is also a water leak sensor: https://templates.blakadder.com/W06.html
